src/main/java/org/broadinstitute/dsde/consent/ontology/resources/TranslateResource.java

```diff
@@ -1,56 +1,80 @@
+  private final TextTranslationService translationService;
+  private final ObjectMapper mapper = new ObjectMapper();
+
+  @Inject
+  public TranslateResource(TextTranslationService translationService) {
+    this.translationService = translationService;
+  }
 
-    @Inject
-    public TranslateResource(TextTranslationService translationService) {
-        this.translationService = translationService;
+  @POST
+  @Path("summary")
+  @Consumes(MediaType.APPLICATION_JSON)
+  public Response translateSummary(String restriction) {
+    try {
+      return Response.ok().entity(translationService.translateDataUseSummary(restriction)).build();
+    } catch (Exception e) {
+      log.error("Error while translating", e);
+      return Response.
+          status(Response.Status.INTERNAL_SERVER_ERROR).
+          entity("Error while translating: " + e.getMessage()).
+          build();
     }
+  }
 
-    @POST
-    @Path("summary")
-    @Consumes(MediaType.APPLICATION_JSON)
-    public Response translateSummary(String restriction) {
-        try {
-            return Response.ok().entity(translationService.translateDataUseSummary(restriction)).build();
-        } catch (Exception e) {
-            log.error("Error while translating", e);
-            return Response.
-                    status(Response.Status.INTERNAL_SERVER_ERROR).
-                    entity("Error while translating: " + e.getMessage()).
-                    build();
-        }
+  @POST
+  @Consumes(MediaType.APPLICATION_JSON)
+  public Response translate(@QueryParam("for") String forParam, String restriction) {
+    try {
+      TranslateFor translateFor = TranslateFor.valueOf(forParam);
+      return buildResponse(translateFor, restriction);
+    } catch (Exception e) {
+      log.error("Error while translating", e);
+      return Response.
+          status(Response.Status.INTERNAL_SERVER_ERROR).
+          entity("Error while translating: " + e.getMessage()).
+          build();
     }
+  }
 
+    @Path("paragraph")
     @POST
-    @Consumes(MediaType.APPLICATION_JSON)
-    public Response translate(@QueryParam("for") String forParam, String restriction) {
-        try {
-            return buildResponse(forParam, restriction);
-        } catch (IOException e) {
-            log.error("Error while translating", e);
-            return Response.
-                    status(Response.Status.INTERNAL_SERVER_ERROR).
-                    entity("Error while translating: " + e.getMessage()).
-                    build();
+    public Response translateParagraph(final String jsonString) throws Exception {
+        final Map<String, String> body = mapper.readValue(jsonString, new TypeReference<>() {});
+        String paragraph = body.get("paragraph");
+
+        if (StringUtils.isBlank(paragraph)) {
+            return Response.status(Response.Status.BAD_REQUEST).entity("Paragraph is required").build();
         }
+
+        return buildResponse(PARAGRAPH, paragraph);
```

Comment:
If you look at the other patterns in this class, we wrap the return in a try so we can control the exact response and log errors appropriately. Something like this would be more informative to both the client and devs reading logs

```
    try {
      return buildResponse(PARAGRAPH, paragraph);
    } catch (Exception e) {
      log.error("Error while translating paragraph: ", e);
      return Response.
          status(Response.Status.INTERNAL_SERVER_ERROR).
          entity("Error while translating paragraph: " + e.getMessage()).
          build();
    }
```